include/zephyr/sensing/sensing_sensor.h

```diff
@@ -93,57 +100,101 @@ struct sensing_connection {
  * build report relationship model base on them, etc.
  */
 struct sensing_sensor {
-	const struct device *dev;
-	const struct sensing_sensor_info *info;
+	const struct device *dev;               /**< Device of the sensor instance. */
+	const struct sensing_sensor_info *info; /**< Info of the sensor instance. */
+	/**< Register info of the sensor instance. */
 	const struct sensing_sensor_register_info *register_info;
-	const uint16_t reporter_num;
-	sys_slist_t client_list;
-	uint32_t interval;
-	uint8_t sensitivity_count;
+	const uint16_t reporter_num; /**< Reporter number of the sensor instance. */
+	sys_slist_t client_list;     /**< Client list of the sensor instance. */
+	uint32_t interval;           /**< Interval of the sensor instance. */
+	uint8_t sensitivity_count;   /**< Sensitivity count of the sensor instance. */
+	/**< Sensitivity of the sensor instance. */
 	int sensitivity[CONFIG_SENSING_MAX_SENSITIVITY_COUNT];
-	enum sensing_sensor_state state;
-	/* runtime info */
-	struct rtio_iodev *iodev;
-	struct k_timer timer;
-	struct rtio_sqe *stream_sqe;
-	atomic_t flag;
-	struct sensing_connection *conns;
+	enum sensing_sensor_state state;  /**< State of the sensor instance. */
+	struct rtio_iodev *iodev;         /**< An RTIO device info of the sensor instance. */
+	struct k_timer timer;             /**< Timer of the sensor instance. */
+	struct rtio_sqe *stream_sqe;      /**< Stream sqe of the sensor instance. */
+	atomic_t flag;                    /**< Flag of the sensor instance. */
+	struct sensing_connection *conns; /**< Connections of the sensor instance. */
 };
 
+/**
+ * @brief Macro to generate a name for a sensor info structure.
+ *
+ * This macro generates a name for a sensor info structure based on a node and an index.
+ */
 #define SENSING_SENSOR_INFO_NAME(node, idx)					\
 	_CONCAT(_CONCAT(__sensing_sensor_info_, idx), DEVICE_DT_NAME_GET(node))
 
+/**
+ * @brief Macro to define a sensor info structure.
+ *
+ * This macro defines a sensor info structure based on a node and an index.
+ * The structure includes the type, name, friendly name, vendor, model, and minimal interval of the
+ * sensor.
+ */
 #define SENSING_SENSOR_INFO_DEFINE(node, idx)				\
 	const static STRUCT_SECTION_ITERABLE(sensing_sensor_info,	\
 			SENSING_SENSOR_INFO_NAME(node, idx)) = {	\
-		.type = DT_PROP_BY_IDX(node, sensor_types, idx),		\
+		.type = DT_PROP_BY_IDX(node, sensor_types, idx),	\
 		.name = DT_NODE_FULL_NAME(node),			\
 		.friendly_name = DT_PROP(node, friendly_name),		\
 		.vendor = DT_NODE_VENDOR_OR(node, NULL),		\
 		.model = DT_NODE_MODEL_OR(node, NULL),			\
-		.minimal_interval = DT_PROP(node, minimal_interval),\
+		.minimal_interval = DT_PROP(node, minimal_interval),	\
 	};
 
+/**
+ * @brief Macro to generate a name for a connections array.
+ *
+ * This macro generates a name for a connections array based on a node.
```

Comment:
would be nice to actually add the `@param`s entries for all the macros

Reply:
Added.